prov/efa/src/rdm/efa_rdm_ep_fiops.c

```diff
@@ -805,23 +812,36 @@ static void efa_rdm_ep_destroy_buffer_pools(struct efa_rdm_ep *efa_rdm_ep)
 		efa_rdm_txe_release(txe);
 	}
 
-	/* Clean up any remaining peers in the hashmap before destroying buffer pools */
-	if (efa_rdm_ep->fi_addr_to_peer_map) {
-		struct efa_rdm_ep_peer_map_entry *map_entry, *map_tmp;
-		HASH_ITER(hndl, efa_rdm_ep->fi_addr_to_peer_map, map_entry, map_tmp) {
-			efa_rdm_peer_destruct(&map_entry->peer, efa_rdm_ep);
-			HASH_DELETE(hndl, efa_rdm_ep->fi_addr_to_peer_map, map_entry);
-			ofi_buf_free(map_entry);
-		}
-	}
+	/* Clean up any remaining peers before destroying buffer pools */
+	dlist_foreach_container_safe (&efa_rdm_ep->ep_peer_list,
+				      struct efa_rdm_peer, peer,
+				      ep_peer_list_entry, tmp) {
+
+		if (peer->conn->fi_addr != FI_ADDR_UNSPEC) {
+			util_av_entry = ofi_bufpool_get_ibuf(
+				efa_rdm_ep->base_ep.av->util_av.av_entry_pool,
+				peer->conn->fi_addr);
 
-	if (efa_rdm_ep->fi_addr_to_peer_map_implicit) {
-		struct efa_rdm_ep_peer_map_entry *map_entry, *map_tmp;
-		HASH_ITER(hndl, efa_rdm_ep->fi_addr_to_peer_map_implicit, map_entry, map_tmp) {
-			efa_rdm_peer_destruct(&map_entry->peer, efa_rdm_ep);
-			HASH_DELETE(hndl, efa_rdm_ep->fi_addr_to_peer_map_implicit, map_entry);
-			ofi_buf_free(map_entry);
+			av_entry = (struct efa_av_entry *) util_av_entry->data;
```

Comment:
Nit: It seems you can move this line before L836 and remove the duplicate one in L833?